generators/typescript-v2/ast/src/ast/core/AstNode.ts

```diff
@@ -6,13 +6,19 @@ export abstract class AstNode extends AbstractAstNode {
     /**
      * Writes the node to a string.
      */
-    public toString(): string {
+    public async toString(): Promise<string> {
+        const writer = new Writer();
+        this.write(writer);
+        return writer.toString();
+    }
```

Comment:
Is there a reason why there would be an async version?
If there's a good reason, I would make a `toStringAsync` rather than a `toStringSync`.
Since all object in JS already have `.toString()` which is sync, I'd keep the sync version as `toString` and if async is necessary, `toStringAsync`.

Reply:
Yea this is on me - we should go back and refactor to prefer the sync method as the `toString` option.